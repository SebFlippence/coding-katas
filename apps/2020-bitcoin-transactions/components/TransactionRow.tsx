import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { Transaction } from '../types/Transaction';
import colors from '../constants/colors';

const categoryIconMap = {
  default: 'file-document-outline',
  'Eating Out': 'food-fork-drink',
  Entertainment: 'movie',
  Gifts: 'gift',
  Groceries: 'cart',
  Subscriptions: 'file-document-box-multiple-outline',
};

export default function TransactionRow({
  id,
  amount,
  counterparty,
  category,
  currency,
  isLastRow,
}: Transaction & { isLastRow: boolean }) {
  return (
    <View
      style={[styles.container, { backgroundColor: colors.rows[id % colors.rows.length] }, isLastRow && styles.lastRow]}
    >
      <Text style={styles.description}>
        <MaterialCommunityIcons
          name={categoryIconMap[category] || categoryIconMap.default}
          title={category}
          size={30}
          style={styles.icon}
        />
        {counterparty}
      </Text>
      <Text>
        {amount} {currency}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: 0,
    borderColor: '#ededed',
  },
  description: { marginTop: -7 },
  lastRow: {
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  icon: { marginRight: 10 },
});
