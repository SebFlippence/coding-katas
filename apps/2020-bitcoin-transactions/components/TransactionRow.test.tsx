import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import TransactionRow from './TransactionRow';

describe('TransactionRow', () => {
  const renderer = new ShallowRenderer();
  const defaultProps = {
    id: 123,
    amount: '1.23',
    counterparty: 'counterparty test',
    currency: 'currency test',
    isLastRow: false,
  };

  test('rendering', () => {
    const tree = renderer.render(<TransactionRow {...defaultProps} category="Bills" />);
    expect(tree).toMatchSnapshot();
  });

  test('rendering last row', () => {
    const tree = renderer.render(<TransactionRow {...defaultProps} category="Gifts" isLastRow />);
    expect(tree).toMatchSnapshot();
  });
});
