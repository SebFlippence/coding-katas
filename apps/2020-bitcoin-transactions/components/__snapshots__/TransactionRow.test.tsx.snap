// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionRow rendering 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#ededed",
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      },
      Object {
        "backgroundColor": "#cecece7d",
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "marginTop": -7,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="file-document-outline"
      size={30}
      style={
        Object {
          "marginRight": 10,
        }
      }
      title="Bills"
    />
    counterparty test
  </Text>
  <Text>
    1.23
     
    currency test
  </Text>
</View>
`;

exports[`TransactionRow rendering last row 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#ededed",
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      },
      Object {
        "backgroundColor": "#cecece7d",
      },
      Object {
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "marginTop": -7,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="gift"
      size={30}
      style={
        Object {
          "marginRight": 10,
        }
      }
      title="Gifts"
    />
    counterparty test
  </Text>
  <Text>
    1.23
     
    currency test
  </Text>
</View>
`;
